src/Diagnostics.ModelsAndUtils/Models/OperationContext.cs

```diff
@@ -7,34 +7,48 @@ namespace Diagnostics.ModelsAndUtils.Models
     /// <summary>
     /// Operation Context
     /// </summary>
-    public class OperationContext
+    public class OperationContext<TResource> where TResource : IResource
     {
         /// <summary>
-        /// Site Resource
+        /// Resource Object
+        /// See <see cref="App"/> and <see cref="HostingEnvironment"/>
         /// </summary>
-        public SiteResource Resource;
+        public TResource Resource { get; private set; }
 
         /// <summary>
         /// Start Time(UTC) for data measurement
         /// </summary>
-        public string StartTime;
+        public string StartTime { get; private set; }
 
         /// <summary>
         /// End Tim(UTC) for data measurement
         /// </summary>
-        public string EndTime;
+        public string EndTime { get; private set; }
+
+        /// <summary>
+        /// sets to false when detector is called from external source (Azure portal, CLI ...)
+        /// sets to true when detexctor is called from internal source (Applens ..)
+        /// </summary>
+        public bool IsInternalCall { get; private set; }
+
+        /// <summary>
+        /// Request Id
+        /// </summary>
+        public string RequestId { get; private set; }
 
         /// <summary>
         /// TimeGrain in minutes for aggregating data.
         /// </summary>
-        public string TimeGrain;
+        public string TimeGrain { get; private set; }
 
```

Comment:
Don't have to add now but in case you want to, we can also add UserAgent here for the near future to distinguish those on the cli tool, applens, prod

Reply:
UserAgent in Operation Context? we should add User Agents in calls made by our backend, right.

Reply:
lets table this for discussion later since its not critical to this change